### ISSUE TITLE
double-click on headers to generate parametric images - doesn't take into account "rotate" nor"scale" 

### ISSUE THREAD

**Issue by romainGuiet** [haesleinhuepf/napari-skimage-regionprops#52]
Hi @haesleinhuepf ,

I saw on the [documentation ](https://www.napari-hub.org/plugins/napari-skimage-regionprops#pluginMetadata) that one can double-click on headers to generate parametric images, that's a wonderful feature! 

But with my layers (that I uploaded with a rotate value)
<img width="1275" alt="image" src="https://user-images.githubusercontent.com/8309560/218443229-c8489945-7196-42ba-8a6d-bb6f397db98f.png">

when double-clicking on a column header , the "measurement image" appears but didn't take into account "rotate" and "scale" from the measured layer nor the labels
<img width="1275" alt="image" src="https://user-images.githubusercontent.com/8309560/218443285-f0a51ef8-c7d8-4743-95ba-d11b5d9b6f49.png">

to reproduce : 
```python
from tifffile import imread # https://pypi.org/project/tifffile/#examples
import napari 
import numpy as np

viewer = napari.Viewer()
viewer.show()

#we define the path 
image_path = '../data/rat_brain/BigStitcher_6-17_441.tif' #3D, 2chs image
image = imread(image_path)

labels_path = '../data/rat_brain/BigStitcher_6-17_441-1-stardist3D_16bit.tif' #labels
labels = imread(labels_path)

print ("image shape : " + str(image.shape))
print ("label shape : " + str(labels.shape))

# we set the voxel scale 
scale = [1,0.625,0.625]
blending = 'additive'
rotate = [0,0,90]

# extract the green channel 
viewer.add_image( np.swapaxes(image, 0,1 ) , 
                    channel_axis = 0,
                    name=["DAPI","Marker" ],
                    colormap=[ 'cyan','green'],
                    scale=scale ,
                    blending=blending,
                    contrast_limits=[[0,15000],[0,15000]],
                    rotate=(rotate,rotate)
                 )

viewer.add_labels( labels,
                    scale=scale ,
                    blending=blending ,
                    rotate=rotate
                 )

```

Cheers,

Romain

**Comment by haesleinhuepf** [comment]
Thanks for the bug report @romainGuiet ! That's easily fixable [here](https://github.com/haesleinhuepf/napari-skimage-regionprops/blob/master/napari_skimage_regionprops/_table.py#L78-L92) by handing over `self._layer.affine` to `new_layer.affine`. We should also do this with `layer.scale`. 

If you have time, an could test this, a PR would be very welcome. Otherwise I'll do this in the coming weeks. No problem.

Thanks again for the feedback! 🤩

**Comment by haesleinhuepf** [comment]
Awesome thanks for your time @romainGuiet ! 🤩

> Can you point me to some tutos / documentations about how to test code modification of a napari plugin from VS Code or PYCharm ?

Not IDE specific, but this page might guide you: https://biapol.github.io/Quantitative_Bio_Image_Analysis_with_Python_2022/day4c_crafting_napari_plugins/04_napari_cookiecutter.html
Also it's very simple: If you navigate in the terminal to the root folder of this repository on your hard drive and call...

```
pip install -e .
```

... the plugin will be installed in your active conda environment. 
Then, you can test it in napari. Writing automated tests for user-interaction is a research field in itself (not joking) and thus, in this case we need to go for manual testing.

**Comment by romainGuiet** [comment]
HI @haesleinhuepf 

>Awesome thanks for your time @romainGuiet !  

Thank you for making all of these plugins and for the link.

I tried, to:
- make a new env
- `pip install napari[all]`
- use `pip install -e .` from the clone&modified folder

but I don't have the menu with `Tools > Measurement Tables > Regionprops (nsr)`

I tried to install `napari-workflow` , and still nothing show up.
Finally I tried to opy the code from "_table.py" in a notebook but then it can't find the package `napari_tools_menu` (while it is installed in this env ) 

I'm sorry for this newbie technical issues 😅 

Cheers

R

**Comment by haesleinhuepf** [comment]
Hey @romainGuiet ,

no worries. I'm full-time hired to support people who struggle with these kind of things. 

> * make a new env
> * `pip install napari[all]`
> * use `pip install -e .` from the clone&modified folder

Suspicious! In such cases I tend to immediately erase my conda installation completely and start from scratch as described here: https://haesleinhuepf.github.io/BioImageAnalysisNotebooks/01_introduction/readme.html

I also install almost everything using mamba ([read more](https://focalplane.biologists.com/2022/12/08/managing-scientific-python-environments-using-conda-mamba-and-friends/)). Try if this works from your base conda environment and from inside the napari-skimage-regionprops folder:
```
mamba create --name devbio-napari-env python=3.9 devbio-napari -c conda-forge
conda activate devbio-napari-env
pip install -e .
```

You're then developing against devbio-napari. The codename of the project is NIJI, because devbio-napari fulfills a similar mission for napari, like Fiji does for ImageJ.

If this doesn't work let's schedule a zoom meeting and troubleshoot together!

Thanks again for your time!

Best,
Robert

**Comment by romainGuiet** [comment]
Hi @haesleinhuepf ,

I installed mambaforge and create a new environment.
I ran `pip install -e .`  (also tried to run the code from a notebook (without import error this time) )
BUT 
The image generated from the column header is still not rotated 😑 😅 
I saw you merged it, did it work for you ? 

BTW in devbio-napari the menu `Tools> Measurements tables >`, I have  `Regionprops (scikit-image nsr)` but there is also `Measurements (n-Simple ITK)` and `Label statistics (clEsperanto)`. 
They all have similar look out, I guess they are also doing the same thing.
Is that necessary to have all of them ? 
I tested all of them and I have the same issue ...

<img width="960" alt="image" src="https://user-images.githubusercontent.com/8309560/221522248-08d39799-204c-4584-abc1-ff74835c513d.png">


Cheers,

Romain

**Comment by haesleinhuepf** [comment]
> I saw you merged it, did it work for you ?

Will check...

> They all have similar look out, I guess they are also doing the same thing.
> Is that necessary to have all of them ?

Well not necessarily. I do find it practical to have measurements from multiple libraries in the same menu. In ImageJ, you find the Particle Analyzer, the ROI manager, the 3D ROI Manager and MorpholibJ's measurements in completely different menus

**Comment by haesleinhuepf** [comment]
Ok, `rotate` was obviously missing in our changes @romainGuiet . Also just to be clear, I think the napari interface with separate `affine`, `scale` and `rotate` is badly designed. If you plan to develop a 3D viewer with layers, don't do it like this ;-)

**Comment by romainGuiet** [comment]
>Ok, rotate was obviously missing in our changes @romainGuiet . 

😅 , indeed it was not in the PR (but I had it in my jupyter notebook, I guess I still need to understand how this works)

I modified ˋ_table.pyˋ to add rotate then ˋpip install -e .ˋ and now the rotate works ! 

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/8309560/221745316-0cb15377-b566-47b4-a654-85dc23132fa6.png">

in 3D with the scale, it's even more amazing  :

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/8309560/221746544-734fd619-51a9-4374-94e5-376fa6152213.png">

**Comment by haesleinhuepf** [comment]
Hey @romainGuiet ,

I just released napari-skimage-regionprops 0.8.2 which addresses this bug. Would you mind confirming that it works for you now?

pip install napari-skimage-regionprops==0.8.2

Thanks again and welcome in [the list of contributors](https://github.com/haesleinhuepf/napari-skimage-regionprops/graphs/contributors)!

Best,
Robert
